tda/streaming.py

```diff
@@ -116,6 +134,109 @@ def req_num(self):
         self.request_number += 1
         return self.request_number
 
+    @classmethod
+    def _relabel_fields(cls, service: Service, msg):
+        '''
+        Takes a raw message, and returns is with relabeled fields.
+        The service must be supported, otherwise we exhibit undefined behaviour
+        '''
+        return cls._field_normalizer(service)(msg)
+
+    @classmethod
+    def _field_normalizer(cls, service: Service):
+        'Returns the field normalizer for a given service.'
+        bfn = cls._book_fieldname_normalizer
+        fn = cls._field_name_normalizer
+
+        known_normalizers = {
+            Service.LISTED_BOOK: bfn(cls.BookFields),
+            Service.NASDAQ_BOOK: bfn(cls.BookFields),
+            Service.OPTIONS_BOOK: bfn(cls.BookFields),
+            Service.NEWS_HEADLINE: bfn(cls.NewsHeadlineFields),
+            Service.TIMESALE_OPTIONS: fn(cls.TimesaleFields),
+            Service.TIMESALE_FUTURES: fn(cls.TimesaleFields),
+            Service.TIMESALE_EQUITY: fn(cls.TimesaleFields),
+            Service.LEVELONE_FUTURES_OPTIONS: fn(cls.LevelOneFuturesFields),
+            Service.LEVELONE_FOREX: fn(cls.LevelOneForexFields),
+            Service.LEVELONE_FUTURES: fn(cls.LevelOneFuturesFields),
+            Service.OPTION: fn(cls.LevelOneOptionFields),
+            Service.QUOTE: fn(cls.LevelOneEquityFields),
+            Service.CHART_FUTURES: fn(cls.ChartFuturesFields),
+            Service.CHART_EQUITY: fn(cls.ChartEquityFields),
+            Service.ACCT_ACTIVITY: fn(cls.AccountActivityFields),
+            # These are services we do not currently handle
+            # so just return the identity function:
+            Service.ACTIVES_NASDAQ: lambda x: x,
+            Service.ADMIN: lambda x: x,
+            Service.ACTIVES_NYSE: lambda x: x,
+            Service.ACTIVES_OTCBB: lambda x: x,
+            Service.ACTIVES_OPTIONS: lambda x: x,
+            Service.FOREX_BOOK: lambda x: x,
+            Service.FUTURES_BOOK: lambda x: x,
+            Service.FUTURES_OPTIONS_BOOK: lambda x: x,
+            Service.CHART_HISTORY_FUTURES: lambda x: x,
+            Service.LEVELTWO_FUTURES: lambda x: x,
+            Service.NEWS_STORY: lambda x: x,
+            Service.NEWS_HEADLINE_LIST: lambda x: x,
+            Service.STREAMER_SERVER: lambda x: x,
+            Service.TIMESALE_FOREX: lambda x: x,
+        }
+
+        return known_normalizers[service]
+
+    @classmethod
+    def _field_name_normalizer(cls, field_enum_type):
+        '''
+        Returns a function which returns a relabeled copy of the argument
+        according to field_enum_type
+        '''
+        def __handler(msg):
+            if 'content' in msg:
+                new_msg = copy.deepcopy(msg)
```

Comment:
Why are you copying msg like this? In python strings are immutable.

Reply:
msg is a dict not a string.

Reply:
Ah, apologies I wrongly assumed that this was passing around a string of JSON like other aspects of the API.